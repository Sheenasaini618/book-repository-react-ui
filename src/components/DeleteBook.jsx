import * as React from 'react';
import Button from '@mui/material/Button';
import Dialog from '@mui/material/Dialog';
import DialogActions from '@mui/material/DialogActions';
import DialogTitle from '@mui/material/DialogTitle';
import DeleteIcon from '@mui/icons-material/Delete';
import IconButton from '@mui/material/IconButton';
import { deleteBook } from '../util/api/bookApi';

export default function DeleteBook(props) {
  const [open, setOpen] = React.useState(false);
  const bookId = props.details

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  const handleDelete = () => {
    deleteBook(bookId).then(res => {
      console.log(res)
      handleClose();
      window.location.reload(false);
    })
  };

  return (
    <div>
      <IconButton className = "icon" onClick={handleClickOpen}>
          <DeleteIcon />
      </IconButton>
      <Dialog
        open={open}
        onClose={handleClose}
        aria-labelledby="responsive-dialog-title"
      >
        <DialogTitle id="responsive-dialog-title">
          {"Are you sure to delete the book?"}
        </DialogTitle>
        <DialogActions>
          <Button className ='closeButton' onClick={handleClose}>
            No
          </Button>
          <Button className ='deleteButton' onClick={handleDelete} autoFocus>
            Yes
          </Button>
        </DialogActions>
      </Dialog>
    </div>
  );
}
