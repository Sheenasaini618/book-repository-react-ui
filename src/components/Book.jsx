import * as React from 'react';
import Card from '@mui/material/Card';
import CardHeader from '@mui/material/CardHeader';
import CardMedia from '@mui/material/CardMedia';
import CardContent from '@mui/material/CardContent';
import CardActions from '@mui/material/CardActions';
import Avatar from '@mui/material/Avatar';
import Typography from '@mui/material/Typography';
import { red } from '@mui/material/colors';
import ManualEditBook from './ManualEditBook';
import DeleteBook from './DeleteBook';


export default function Book(props) {

  var book = props.details
  console.log(book)
  return (
    <Card sx={{ maxWidth: 450, maxHeight: 500 }} style={{ marginTop: "10px" }}>
      <CardHeader
        avatar={
          <Avatar sx={{ bgcolor: red[500] }} aria-label="recipe">
            {book.title.charAt(0).toUpperCase()}
          </Avatar>
        }
        title={book.title}
        subheader={book.authors[0]}
      />
      <CardMedia
        component="img"
        sx={{ width: 151 }}
        height="194"
        image={book.imageLinks.smallThumbnail}
        alt="Paella dish"
      />
      <CardContent>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Price: â‚¹{book.price}.00
        </Typography>
        <Typography variant="h6" color="text.secondary" align='justify' style={{ fontWeight: 600 }}>
          Copies Available: {book.quantity}
        </Typography>
      </CardContent>
      <CardActions disableSpacing>
        <ManualEditBook details={book} />
        <DeleteBook details={book.id} />
      </CardActions>
    </Card>
  );
}
