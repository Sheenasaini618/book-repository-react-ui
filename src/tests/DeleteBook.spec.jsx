import React from 'react'
import Book from '../components/Book'
import '@testing-library/jest-dom'
import Adapter from 'enzyme-adapter-react-16';
import { mount, shallow, render } from 'enzyme'
import { configure } from 'enzyme';
import DeleteBook from '../components/DeleteBook';


configure({ adapter: new Adapter() });

global.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve({
        ok: true,
        status: 200,
    }),
);
describe('<DeleteBook />', () => {

    let props, wrapper;

    beforeEach(() => {
        props = {
            details: 1
        };
    })

    it('should check for icon element', () => {

        wrapper = shallow(<DeleteBook {...props} />);
        const event = {};
        expect(wrapper.find('.icon').length).toBe(1);
        wrapper
            .find('.icon')
            .at(0)
            .simulate('click', event);
    })
})

describe('<DeleteBook />', () => {

    let props, component, wrapper;

    beforeEach(() => {
        props = {
            details: []
        };

        component = shallow(<DeleteBook {...props} />);
    });

    if (typeof window === 'undefined') {
        global.window = {}
    }

    it('should match snapshot', () => {
        const element = component.debug()
        expect(element).toMatchSnapshot();
    });

    it('should validate elements in delete book component', () => {

        wrapper = shallow(<DeleteBook {...props} />);
        const event = {};
        expect(wrapper.find('.icon').length).toBe(1);
        wrapper.find('.closeButton').at(0).simulate('click', event);
        wrapper.find('.deleteButton').at(0).simulate('click', event);
    })
})