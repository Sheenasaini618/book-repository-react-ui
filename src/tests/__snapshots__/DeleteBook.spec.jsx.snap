// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Book /> should render book  1`] = `
"<ForwardRef(Card) sx={{...}} style={{...}}>
  <ForwardRef(CardHeader) avatar={{...}} title=\\"Probability Basics\\" subheader=\\"Michael Biden\\" />
  <ForwardRef(CardMedia) component=\\"img\\" sx={{...}} height=\\"194\\" image=\\"http://books.google.com/books/content?id=3yq_oQEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api\\" alt=\\"Paella dish\\" />
  <ForwardRef(CardContent)>
    <ForwardRef(Typography) variant=\\"h6\\" color=\\"text.secondary\\" align=\\"justify\\" style={{...}}>
      Price: â‚¹
      1000
      .00
    </ForwardRef(Typography)>
    <ForwardRef(Typography) variant=\\"h6\\" color=\\"text.secondary\\" align=\\"justify\\" style={{...}}>
      Copies Available: 
      8
    </ForwardRef(Typography)>
  </ForwardRef(CardContent)>
  <ForwardRef(CardActions) disableSpacing={true}>
    <ManualEditBook details={{...}} />
    <DeleteBook details=\\"1\\" />
  </ForwardRef(CardActions)>
</ForwardRef(Card)>"
`;

exports[`<DeleteBook /> should render book  1`] = `
"<div>
  <ForwardRef(IconButton) className=\\"icon\\" onClick={[Function: handleClickOpen]}>
    <Memo(ForwardRef(DeleteIcon)) />
  </ForwardRef(IconButton)>
  <ForwardRef(Dialog) open={false} onClose={[Function: handleClose]} aria-labelledby=\\"responsive-dialog-title\\">
    <ForwardRef(DialogTitle) id=\\"responsive-dialog-title\\">
      Are you sure to delete the book?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button) onClick={[Function: handleClose]}>
        No
      </ForwardRef(Button)>
      <ForwardRef(Button) onClick={[Function: handleDelete]} autoFocus={true}>
        Yes
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</div>"
`;
